react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx

```diff
@@ -68,6 +72,24 @@ export const useFdm3dNodeData = (
   return result;
 };
 
+export const useFdmAssetMappings = (
+  fdmAssetExternalIds: DmsUniqueIdentifier[],
+  models: TypedReveal3DModel[]
+): UseQueryResult<ThreeDModelMappings[]> => {
+  const nodeCacheContent = useContext(FdmNodeCacheContext);
+
+  return useQuery(
```

Comment:
Isn't pagination needed for FDM asset mapping?

Reply:
It fetches all of them at once under the hood (`filterAllInstances` in `FdmSDK.ts`). I ended up doing it this way because it makes the cache less complicated.. Otherwise I'd have to keep track of whether all edges have been registered for a tree index or not when getting edges for a tree index.